Client/Frontend/Shields/ShareShieldsController.swift

```diff
@@ -0,0 +1,278 @@
+
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = Strings.Share.shareScreenTitle
+
+        doLayoutAndAddGestureRecognizers()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        preferredContentSize = contentStackView.systemLayoutSizeFitting(
+            CGSize(width: view.bounds.size.width - UX.contentSizeChange,
+                   height: view.bounds.size.height - UX.contentSizeChange),
+            withHorizontalFittingPriority: .required,
+            verticalFittingPriority: .fittingSizeLevel
+        ).with {
+            $0.width += UX.contentSizeChange
+        }
+
+        navigationController?.preferredContentSize = preferredContentSize
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if #available(iOS 13.0, *) {
+            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+                applyTheme(theme)
+            }
+        }
+    }
+
+    // MARK: Internal
+    
+    private func doLayoutAndAddGestureRecognizers() {
+        let shareTypeViewEmail = ShareTypeView(type: .email)
+        let shareEmailGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewEmail.addGestureRecognizer(shareEmailGesture)
+
+        let shareTypeViewTwitter = ShareTypeView(type: .twitter)
+        let shareTwitterGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewTwitter.addGestureRecognizer(shareTwitterGesture)
+
+        let shareTypeViewFacebook = ShareTypeView(type: .facebook)
+        let shareFacebookGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewFacebook.addGestureRecognizer(shareFacebookGesture)
+
+        let shareTypeViewDefault = ShareTypeView(type: .default)
+        let shareDefaultGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewDefault.addGestureRecognizer(shareDefaultGesture)
+
+        view.addSubview(contentStackView)
+
+        contentStackView.snp.makeConstraints {
+            $0.top.leading.trailing.equalTo(view.safeAreaLayoutGuide)
+        }
+
+        contentStackView.addStackViewItems(
+            .view(shareTypeViewEmail),
+            .view(shareTypeViewTwitter),
+            .view(shareTypeViewFacebook),
+            .view(shareTypeViewDefault)
+        )
+    }
+
+    // MARK: Themeable
+    
+    func applyTheme(_ theme: Theme) {
+        view.backgroundColor = theme.isDark ? BraveUX.popoverDarkBackground : UIColor.white
+
+        contentStackView.arrangedSubviews.forEach { subview in
+            if let typeView = subview as? ShareTypeView {
```

Comment:
Probably best to just check if its `Themable` rather than it specifically being a `ShareTypeView`